server/base/src/main/java/org/apache/accumulo/server/util/fateCommand/FateTxnDetails.java

```diff
@@ -100,6 +113,14 @@ public String getStatus() {
     return status;
   }
 
+  public List<String> getLocksHeld() {
+    return Collections.unmodifiableList(locksHeld);
```

Comment:
`List.of` returns an unmodifiable List, I don't think we need to wrap it again in the case that it's not set. Instead of initializing the `locksHeld` and `locksWaiting` variables with `List.of`, you could mark them final and initialize them in the constructor with an unmodifiable list.